wasm-beep example fails to compile

I get these messages on MacOS Monterey, rustc v1.65, compiling via `npm start` or `npm run build`


```bash
   Compiling wasm-beep v0.1.0 (/Users/gwiley/Working/cpal/examples/wasm-beep)
error[E0277]: the trait bound `T: SizedSample` is not satisfied
   --> src/lib.rs:61:10
    |
61  |         .build_output_stream(
    |          ^^^^^^^^^^^^^^^^^^^ the trait `SizedSample` is not implemented for `T`
    |
note: required by a bound in `build_output_stream`
   --> /Users/gwiley/Working/cpal/src/traits.rs:161:12
    |
161 |         T: SizedSample,
    |            ^^^^^^^^^^^ required by this bound in `build_output_stream`
help: consider further restricting this bound
    |
46  |     T: cpal::Sample + cpal::SizedSample,
    |                     +++++++++++++++++++

error[E0782]: trait objects must include the `dyn` keyword
  --> src/lib.rs:77:24
   |
77 |         let value: T = cpal::Sample::from::<f32>(&next_sample());
   |                        ^^^^^^^^^^^^
   |
help: add `dyn` keyword before this trait
   |
77 |         let value: T = <dyn cpal::Sample>::from::<f32>(&next_sample());
   |                        ++++             +

error[E0038]: the trait `Sample` cannot be made into an object
  --> src/lib.rs:77:24
   |
77 |         let value: T = cpal::Sample::from::<f32>(&next_sample());
   |                        ^^^^^^^^^^^^ `Sample` cannot be made into an object
   |
note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
  --> /Users/gwiley/.cargo/registry/src/github.com-1ecc6299db9ec823/dasp_sample-0.11.0/src/lib.rs:39:34
   |
39 | pub trait Sample: Copy + Clone + PartialOrd + PartialEq {
   |                                  ^^^^^^^^^^   ^^^^^^^^^ the trait cannot be made into an object because it uses `Self` as a type parameter
   |                                  |
   |                                  the trait cannot be made into an object because it uses `Self` as a type parameter

Some errors have detailed explanations: E0038, E0277, E0782.
For more information about an error, try `rustc --explain E0038`.
error: could not compile `wasm-beep` due to 3 previous errors
```

--- a/.github/workflows/cpal.yml
+++ b/.github/workflows/cpal.yml
@@ -313,8 +313,8 @@ jobs:
     - name: Build iphonesimulator feedback example
       run: cd examples/ios-feedback && xcodebuild -scheme cpal-ios-example -configuration Debug -derivedDataPath build -sdk iphonesimulator
 
-  wasm-beep-check:
-    # this only checks that the Rust source builds
+  wasm-beep-build:
+    # this only confirms that the Rust source builds
     # and checks to prevent regressions like #721.
     #
     # It does not test the javascript/web integration
@@ -323,5 +323,5 @@ jobs:
       - uses: actions/checkout@v2
       - name: Cargo Check
         working-directory: ./examples/wasm-beep
-        run: cargo check
+        run: cargo build --example beep --target wasm32-unknown-unknown --features=wasm-bindgen
 

--- a/.github/workflows/cpal.yml
+++ b/.github/workflows/cpal.yml
@@ -321,7 +321,9 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - name: Cargo Check
+      - name: Install Target
+        run: rustup taret add wasm32-unknown-unknown
+      - name: Cargo Build
         working-directory: ./examples/wasm-beep
         run: cargo build --target wasm32-unknown-unknown
 

--- a/.github/workflows/cpal.yml
+++ b/.github/workflows/cpal.yml
@@ -313,3 +313,15 @@ jobs:
     - name: Build iphonesimulator feedback example
       run: cd examples/ios-feedback && xcodebuild -scheme cpal-ios-example -configuration Debug -derivedDataPath build -sdk iphonesimulator
 
+  wasm-beep-check:
+    # this only checks that the Rust source builds
+    # and checks to prevent regressions like #721.
+    #
+    # It does not test the javascript/web integration
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cargo Check
+        working-directory: ./examples/wasm-beep
+        run: cargo check
+

--- a/.github/workflows/cpal.yml
+++ b/.github/workflows/cpal.yml
@@ -323,5 +323,5 @@ jobs:
       - uses: actions/checkout@v2
       - name: Cargo Check
         working-directory: ./examples/wasm-beep
-        run: cargo build --example beep --target wasm32-unknown-unknown --features=wasm-bindgen
+        run: cargo build --target wasm32-unknown-unknown
 

--- a/.github/workflows/cpal.yml
+++ b/.github/workflows/cpal.yml
@@ -322,7 +322,7 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - name: Install Target
-        run: rustup taret add wasm32-unknown-unknown
+        run: rustup target add wasm32-unknown-unknown
       - name: Cargo Build
         working-directory: ./examples/wasm-beep
         run: cargo build --target wasm32-unknown-unknown

--- a/examples/wasm-beep/src/lib.rs
+++ b/examples/wasm-beep/src/lib.rs
@@ -43,7 +43,7 @@ pub fn beep() -> Handle {
 
 fn run<T>(device: &cpal::Device, config: &cpal::StreamConfig) -> Stream
 where
-    T: cpal::Sample,
+    T: cpal::Sample + cpal::SizedSample,
 {
     let sample_rate = config.sample_rate.0 as f32;
     let channels = config.channels as usize;

--- a/examples/wasm-beep/src/lib.rs
+++ b/examples/wasm-beep/src/lib.rs
@@ -43,7 +43,7 @@ pub fn beep() -> Handle {
 
 fn run<T>(device: &cpal::Device, config: &cpal::StreamConfig) -> Stream
 where
-    T: cpal::Sample + cpal::SizedSample,
+    T: cpal::Sample + cpal::SizedSample + cpal::FromSample<f32>,
 {
     let sample_rate = config.sample_rate.0 as f32;
     let channels = config.channels as usize;
@@ -71,10 +71,11 @@ where
 
 fn write_data<T>(output: &mut [T], channels: usize, next_sample: &mut dyn FnMut() -> f32)
 where
-    T: cpal::Sample,
+    T: cpal::Sample + cpal::FromSample<f32>,
 {
     for frame in output.chunks_mut(channels) {
-        let value: T = <dyn cpal::Sample>::from::<f32>(&next_sample());
+        let sample = next_sample();
+        let value = T::from_sample::<f32>(sample);
         for sample in frame.iter_mut() {
             *sample = value;
         }

--- a/examples/wasm-beep/src/lib.rs
+++ b/examples/wasm-beep/src/lib.rs
@@ -74,7 +74,7 @@ where
     T: cpal::Sample,
 {
     for frame in output.chunks_mut(channels) {
-        let value: T = cpal::Sample::from::<f32>(&next_sample());
+        let value: T = <dyn cpal::Sample>::from::<f32>(&next_sample());
         for sample in frame.iter_mut() {
             *sample = value;
         }

--- a/examples/wasm-beep/src/lib.rs
+++ b/examples/wasm-beep/src/lib.rs
@@ -38,6 +38,7 @@ pub fn beep() -> Handle {
         cpal::SampleFormat::F32 => run::<f32>(&device, &config.into()),
         cpal::SampleFormat::I16 => run::<i16>(&device, &config.into()),
         cpal::SampleFormat::U16 => run::<u16>(&device, &config.into()),
+        _ => panic!("unsupported sample format"),
     })
 }
 

